temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/properties/TemporalProperties.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.spring.boot.autoconfigure.properties;
+
+import java.util.List;
+import javax.annotation.Nullable;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.ConstructorBinding;
+import org.springframework.boot.context.properties.NestedConfigurationProperty;
+
+@ConfigurationProperties(prefix = "spring.temporal")
```

Comment:
Can you think of a use case to wrap all of these properties as a collection and bring this one level higher? (or is there a layer in Spring that lets that already happen?)

For example, what if I want to auto configure two workers with separate clients? With this setup, can I have two workflows on one namespace and two other workflows on another namespace?

Alternatively we could think about making clients and workers accept namespaces so I could provide `namespaces` alongside `taskQueues` in my `@WorkflowImpl` (or maybe `workerName` in the annotation instead, unsure). But meh, we can stick to not supporting same-task-queue-different-namespace.

Reply:
I thought about it a little bit and it's an interesting question.  I didn't find an elegant solution yet and I'm gonna tackle it in a separate PR.

Here is the problem. We have too many moving parts. 

We have namespaces. So, ideally, we should allow setting this whole config per-namespace (collection as you said). WorkflowImpl and ActivityImpl will have to have a pair namespace:taskQueue for auto-discovery after that. Things become a bit more complex, but in a manageable manner. This is on me to-do list of next things.

Multiple clients. I think it's not really worth the complexity. I do have a to-do in mind on supporting separate client and worker clients in the next PR. Now we are cheating and use just one for both, but it's not the correct approach. We should have a separate workerClient and well, clientClient. I probably shouldn't do this part in this PR to establish a correct structure, but I expect it to be a messy and frustrating task of its own and it will be in a separate PR. To do it well, I need to come up with a way for worker and client Clients to share the majority of the config.